### PR TITLE
Longest repeating subsequence in C++, Java and Python #844

### LINKED ISSUE #844: Longest Repeating Subsequence in C++, Java and Python
To find the longest repeating subsequence in C++, Java, and Python

Can I work on this?

### PR DESCRIPTION
**Pull request by prashant-mahanta**
Fixes #844  

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

-  Longest Repeating Subsequence using Dynamic Programming

#### Languages Used:
-  C++
-  Java
- Python

#### Files Added:
-  Longest_Repeating_Subsequence.cpp
-  Longest_Repeating_Subsequence.py
- Longest_Repeating_Subsequence.java

### CODE REVIEW

File: Longest_Repeating_Subsequence/Longest_Repeating_Subsequence.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#include<bits/stdc++.h>
+
+using namespace std;
+
+int LongestRepeatingSubsequence(string str){
```
Comment (marwahmanbir) [inline Longest_Repeating_Subsequence/Longest_Repeating_Subsequence.cpp]: Add a space after ``{``

File: Longest_Repeating_Subsequence/Longest_Repeating_Subsequence.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    int length;
+    length = str.length();
+    //Initialize all dp elements with 0
```
Comment (marwahmanbir) [inline Longest_Repeating_Subsequence/Longest_Repeating_Subsequence.cpp]: A single blank line should be there before every single-line comment. Also add a space after ``//`` in every comment.

File: Longest_Repeating_Subsequence/Longest_Repeating_Subsequence.cpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    vector<vector<int>> dp(length+1, vector<int>(length+1, 0));
+
+    for(int i=1; i<=length; i++){
```
Comment (marwahmanbir) [inline Longest_Repeating_Subsequence/Longest_Repeating_Subsequence.cpp]: Add single space after each ``for``.

Comment (marwahmanbir) [reply Longest_Repeating_Subsequence/Longest_Repeating_Subsequence.cpp]: Also add a single space before ``{``.

**Comment by marwahmanbir** [comment]
Consider the changes for whole code and not only the lines they're mentioned against.

**Comment by prashant-mahanta** [comment]
@marwahmanbir I have fixed the mentioned changes. Please have a look at it.

**Comment by marwahmanbir** [comment]
I see that you've fixed the comments but I also told you to add a single space before ``{`` and not after it.

**Comment by marwahmanbir** [comment]
And the changes mentioned were for both C++ and Java files.

**Comment by prashant-mahanta** [comment]
@marwahmanbir I have fixed the requested changes. Have a look at it.

**Comment by marwahmanbir** [comment]
Good work. :+1:

**Pull request merged**